text/0000-named-tracers.md

```diff
@@ -0,0 +1,104 @@
+# Named Tracers
+
+**Status:** `proposed`
+
+_Creating Tracers using a factory mechanism and naming those Tracers in accordance with the library that provides the instrumentation for a traced component._
+
+## Suggested reading
+
+* [Proposal: Tracer Components](https://github.com/open-telemetry/opentelemetry-specification/issues/10)
+* [Global Instance discussions](https://github.com/open-telemetry/opentelemetry-specification/labels/global%20instance)
+* [Proposal: Add a version resource](https://github.com/open-telemetry/oteps/pull/38)
+
+## Motivation
+
+The mechanism of "Named Tracers" proposed here is motivated by following scenarios:
+
+* For a consumer of OpenTelemetry instrumentation libraries, there is currently no possibility of influencing the amount of the data produced by such libraries. Instrumentation libraries can easily "spam" backend systems, deliver bogus data or - in the worst case - crash or slow down applications. These problems might even occur suddenly in production environments caused by external factors such as increasing load or unexpected input data.
```

Comment:
I am confused a bit by the usage of " instrumentation library". Does it refer to a 3rd party library that is instrumented using OpenTelemetry API?
If so I would change the wording to be "instrumented library". "Instrumentation library" IMO is a better label for the OpenTelemetry library itself and is confusing in this context.

Reply:
We explicitly don't mean the instrumented library here, but the library that does the instrumentation (I've seen them called integration), whether it is a 3rd party or developed in an OpenTracing repository, or even some auto-injected code from an Agent (see Zero Source Code Modification Instrumentation OTEP). See also the examples for identifiers at the end.

Reply:
OK, just to double check an "instrumentation library"  in this doc refers to a library that adds/injects telemetry calls to another library or otherwise helps that library to make telemetry calls.

Is that distinction important for the purposes of this document? I would imagine "Named Tracers" could equally well be directly used by the instrumented library. I think the fact that the instrumented library and the library that instruments it are separate is an orthogonal concept and can be left out from this discussion.

I may be missing something so please correct me.

Reply:
If a library (or even application) has instrumentation built-in, it is of course both the instrumenting and instrumented library and can pass its own name here. In all other cases (and to distinguish them from that case), the distinction between instrumenting and instrumented library is very important. For example, if a HTTP library com.example.http is instrumented by either io.opentelemetry.contrib.examplehttp or com.example.company.examplehttpintegration, then it is important that the tracer is *not* named com.example.http but either io.opentelemetry.contrib.examplehttp or com.example.company.examplehttpintegration.

Reply:
OK, makes sense, it was not apparent from the document. Perhaps worth adding a few words explaining what the "name" is intended to be.

Reply:
I updated the "Explanation" section and tried to clarify the "name" part.

Reply:
LGTM now, thanks.

Reply:
Could you approve the proposal, please?